### PR TITLE
[567448] validation duplicate combined fragment

### PR DESCRIPTION
**Pull request by MalinaStefaniaStoicanescu**
Change-Id: Ieae93ec6378b4bb84f15ea99598c19c3e5ac3757
Signed-off-by: MalinaStefaniaStoicanescu <malina.stoicanescu@thalesgroup.com>

### CODE REVIEW

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend [hunk 1]
```diff
@@ -157,29 +157,34 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 		checkDuplicated(message, model as Model, newHashSet)	
 	}
 	
-	def boolean checkDuplicated(Message message, EObject model, HashSet<String> names) {
+	def boolean checkDuplicated(EObject elementToCheck, EObject model, HashSet<String> names) {
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:160]: Add a separate function to check duplicated combined fragments.
Do not edit in the same place as for messages, let this checkDuplicated unchanged.

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend [hunk 2]
```diff
@@ -346,6 +351,11 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 		}
 	}
 	
+	@Check
+	def checkDuplicatedCombinedFragment(CombinedFragment combinedFragment) {
+		checkDuplicated(combinedFragment, TextualScenarioHelper.getModelContainer(combinedFragment), newHashSet)
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:357]: Put here the code from check duplicated that is specific to combined fragments.

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend [hunk 3]
```diff
@@ -726,13 +736,22 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 		p.name + ":" + p.keyword
 	}
 
-	def getMessagesMapKey(EObject message) {
-		if (message instanceof SequenceMessageType) {
-			return message.name + ":" + message.arrow + ":" + message.source + ":" + message.target
+	def getMessagesMapKey(EObject element) {
+
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:753]: Do not add in the getMessagesKeys the combined fragments.

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend [hunk 4]
```diff
@@ -157,29 +157,35 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
...
+					} else if (element instanceof ArmTimerMessage) {
 						error('The same exchange is already used in text editor on timeline \"' +
-							(element as ArmTimerMessage).participant +"\"!",
+							(element as ArmTimerMessage).participant + "\"!",
 							TextualScenarioPackage.Literals.MESSAGE__NAME)
+					} else if (element instanceof CombinedFragment) {
+						error(String.format(
+							"The same " + element.keyword + " with expression \" " + element.expression +
+								"\" and timelines " + element.timelines.toSet.sort + " is already used in text editor!"
+						), TextualScenarioPackage.Literals.COMBINED_FRAGMENT__EXPRESSION)
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:181]: No need to sort, just element.timelines.

Comment (MalinaStefaniaStoicanescu) [reply plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:181]: done

**Review by georgiana-ecobici — APPROVED** [review]

**Pull request merged**